### PR TITLE
[Documentation] Architecture - Resource Layer & State Machine

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | yes
| New docs?    | yes
| BC breaks?      | no
| Related tickets | partially #5275 
| License         | MIT

### CODE REVIEW

File: docs/book/architecture.rst [hunk 1]
```diff
@@ -17,6 +17,11 @@ Resource Layer
 We created an abstraction on top of Doctrine, in order to have a consistent and flexible way to manage all the resources. By "resource" we understand every model in the application.
 Simplest examples of Sylius resources are "product", "order", "tax_category", "promotion", "user", "shipping_method" and so on...
 
+There are two types of resources in **Sylius**:
+
+* registered by default - their names begin with ``sylius_*`` for example: ``sylius_product``
```
Comment (pjedrzejewski) [inline docs/book/architecture.rst]: Actually, aliases are ``sylius.product``, we use dots to separate application name from the resource name.

File: docs/book/state_machine.rst [hunk 2]
```diff
@@ -4,34 +4,97 @@
...
+State Machines are an approach to handling changes occurring in the system frequently, that is extremely flexible and very well organised.
+
+Every state machine will have a predefined set of states, that will be stored on an entity that is being controlled by it.
+These states will have a set of defined transitions between them, and a set of callbacks - a kind of events, that will happen on defined transitions.
 
 States
 ------
 
-...
+States of a state machine are defined as constatnts on the model of an entity that the state machine is controlling.
```
Comment (pjedrzejewski) [inline docs/book/state_machine.rst]: constatnts -> constants

File: docs/book/architecture.rst [hunk 3]
```diff
@@ -17,6 +17,11 @@ Resource Layer
...
+* custom resources, from your application which have a separate convention. We place them under ``sylius_resource:`` ``resource_name:`` in the ``config.yml``.
```
Comment (lchrusciel) [inline docs/book/architecture.rst]: We should add a note, that we recommend to use an `app.*` convention with custom resources.

**Comment by pjedrzejewski** [comment]
Thank you Magda!

File: docs/book/architecture.rst [hunk 4]
```diff
@@ -17,6 +17,11 @@ Resource Layer
 We created an abstraction on top of Doctrine, in order to have a consistent and flexible way to manage all the resources. By "resource" we understand every model in the application.
 Simplest examples of Sylius resources are "product", "order", "tax_category", "promotion", "user", "shipping_method" and so on...
 
+There are two types of resources in **Sylius**:
+
+* registered by default - their names begin with ``sylius.*`` for example: ``sylius.product``
+* custom resources, from your application which have a separate convention. We place them under ``sylius_resource:`` ``resource_name:`` in the ``config.yml``. For these we reccomend using the naming convenion of ``app.*`` for instance ``app.my_entity``.
```
Comment (lchrusciel) [inline docs/book/architecture.rst]: `reccomend` -> `recommend`

**Pull request merged**